drivers/adc/adc_mcux_adc12.c

```diff
@@ -254,134 +254,49 @@ static const struct adc_driver_api mcux_adc12_driver_api = {
-static void mcux_adc12_config_func_1(struct device *dev);
-
-ASSERT_WITHIN_RANGE(DT_INST_PROP(1, clk_source), 0, 3,
-		    "Invalid clock source");
-ASSERT_ADC12_CLK_DIV_VALID(DT_INST_PROP(1, clk_divider),
-			   "Invalid clock divider");
-ASSERT_WITHIN_RANGE(DT_INST_PROP(1, sample_time), 2, 256,
-		    "Invalid sample time");
-static const struct mcux_adc12_config mcux_adc12_config_1 = {
-	.base = (ADC_Type *)DT_INST_REG_ADDR(1),
-	.clock_src = TO_ADC12_CLOCK_SRC(DT_INST_PROP(1, clk_source)),
-	.clock_div =
-		TO_ADC12_CLOCK_DIV(DT_INST_PROP(1, clk_divider)),
-#if DT_INST_PROP(1, alternate_voltage_reference) == 1
-	.ref_src = kADC12_ReferenceVoltageSourceValt,
-#else
-	.ref_src = kADC12_ReferenceVoltageSourceVref,
-#endif
-	.sample_clk_count = DT_INST_PROP(1, sample_time),
-	.irq_config_func = mcux_adc12_config_func_1,
-};
-
-static struct mcux_adc12_data mcux_adc12_data_1 = {
-	ADC_CONTEXT_INIT_TIMER(mcux_adc12_data_1, ctx),
-	ADC_CONTEXT_INIT_LOCK(mcux_adc12_data_1, ctx),
-	ADC_CONTEXT_INIT_SYNC(mcux_adc12_data_1, ctx),
-};
-
-DEVICE_AND_API_INIT(mcux_adc12_1, DT_INST_LABEL(1),
-		    &mcux_adc12_init, &mcux_adc12_data_1, &mcux_adc12_config_1,
-		    POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEVICE,
-		    &mcux_adc12_driver_api);
-
-static void mcux_adc12_config_func_1(struct device *dev)
-{
-	IRQ_CONNECT(DT_INST_IRQN(1),
-		    DT_INST_IRQ(1, priority), mcux_adc12_isr,
-		    DEVICE_GET(mcux_adc12_1), 0);
-
-	irq_enable(DT_INST_IRQN(1));
-}
-#endif /* DT_HAS_DRV_INST(1) */
-
-#if DT_HAS_DRV_INST(2)
-static void mcux_adc12_config_func_2(struct device *dev);
-
-ASSERT_WITHIN_RANGE(DT_INST_PROP(2, clk_source), 0, 3,
-		    "Invalid clock source");
-ASSERT_ADC12_CLK_DIV_VALID(DT_INST_PROP(2, clk_divider),
-			   "Invalid clock divider");
-ASSERT_WITHIN_RANGE(DT_INST_PROP(2, sample_time), 2, 256,
-		    "Invalid sample time");
-static const struct mcux_adc12_config mcux_adc12_config_2 = {
-	.base = (ADC_Type *)DT_INST_REG_ADDR(2),
-	.clock_src = TO_ADC12_CLOCK_SRC(DT_INST_PROP(2, clk_source)),
-	.clock_div =
-		TO_ADC12_CLOCK_DIV(DT_INST_PROP(2, clk_divider)),
-#if DT_INST_PROP(2, alternate_voltage_reference) == 1
-	.ref_src = kADC12_ReferenceVoltageSourceValt,
-#else
-	.ref_src = kADC12_ReferenceVoltageSourceVref,
-#endif
-	.sample_clk_count = DT_INST_PROP(2, sample_time),
-	.irq_config_func = mcux_adc12_config_func_2,
-};
-
-static struct mcux_adc12_data mcux_adc12_data_2 = {
-	ADC_CONTEXT_INIT_TIMER(mcux_adc12_data_2, ctx),
-	ADC_CONTEXT_INIT_LOCK(mcux_adc12_data_2, ctx),
-	ADC_CONTEXT_INIT_SYNC(mcux_adc12_data_2, ctx),
-};
-
-DEVICE_AND_API_INIT(mcux_adc12_2, DT_INST_LABEL(2),
-		    &mcux_adc12_init, &mcux_adc12_data_2, &mcux_adc12_config_2,
-		    POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEVICE,
-		    &mcux_adc12_driver_api);
-
-static void mcux_adc12_config_func_2(struct device *dev)
-{
-	IRQ_CONNECT(DT_INST_IRQN(2),
-		    DT_INST_IRQ(2, priority), mcux_adc12_isr,
-		    DEVICE_GET(mcux_adc12_2), 0);
+#define ADC12_REF_SCR(n)						\
```

Comment:
Did you mean to call the macro `ADC12_REF_SCR` or was it a typo? `ADC12_REF_SRC`?

Reply:
typo, will fix.

Reply:
this still needs to be done

Reply:
I swore I did that, oh well. done now :)